src/main/kotlin/com/cognifide/gradle/aem/common/file/resolver/Resolver.kt

```diff
@@ -56,8 +68,19 @@ val downloadDir: File
         aem.progress {
             step = "Resolving files"
             total = size.toLong()
-            aem.parallel.each(this@apply) { group ->
-                increment("Group '${group.name}'") { group.resolve() }
+
+            if (parallelLevel <= 1) {
+                forEach { group ->
+                    increment("Group '${group.name}'") { group.resolve() }
+                }
+            } else {
+                val (parallel, sequential) = partition { it.parallelable }
+                aem.parallel.poolEach(parallelLevel, "resolver", parallel) { group ->
+                    increment("Group '${group.name}'") { group.resolve() }
+                }
+                sequential.forEach { group ->
+                    increment("Group '${group.name}'") { group.resolve() }
```

Comment:
detachedConfiguration on thread different than main Gradle thread... cannot be resolved in Gradle 6.
https://docs.gradle.org/5.6.4/userguide/troubleshooting_dependency_resolution.html#sub:configuration_resolution_constraints